FlakyTest: tests/chainerx_tests/unit_tests/routines_tests/test_misc.py::test_Maximum

[Travis](https://travis-ci.org/chainer/chainer/jobs/574765416)

```test_Maximum_param_174_{in_shapes=((3, 1), (3, 4)), in_dtypes=('float16', 'float16'), input_lhs='random', is_module=False, input_rhs='random', out_dtype='float16'}[native:0]```

```
self = <chainer.gradient_check._CheckBackward object at 0x7f206a88bda0>
gx_numeric = array(-0.26232969, shape=(), dtype=float64, device='native:0')
gx_backward = array(-0.28092024, shape=(), dtype=float64, device='native:0')
directions = [array([[0.43320103],
       [-0.14267083],
       [0.18341649]], shape=(3, 1), dtype=float64, device='native:0'), arr....01276379],
       [-0.35331611, 0.20300687, 0.09588751, 0.10208457]], shape=(3, 4), dtype=float64, device='native:0')]
    def _compare_gradients(self, gx_numeric, gx_backward, directions):
        atol = self.atol
        rtol = self.rtol
        # Compare the gradients
        try:
            testing.assert_allclose(
                gx_numeric, gx_backward, atol=atol, rtol=rtol)
        except AssertionError as e:
            eps = self.eps
            xs = self.xs
            gys = self.gys
            f = six.StringIO()
            f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
                eps, atol, rtol))
            for i, x in enumerate(xs):
                f.write('inputs[{}]:\n'.format(i))
                f.write('{}\n'.format(x))
            for i, gy in enumerate(gys):
                f.write('grad_outputs[{}]:\n'.format(i))
                f.write('{}\n'.format(gy))
            for i, d in enumerate(directions):
                f.write('directions[{}]:\n'.format(i))
                f.write('{}\n'.format(d))
            f.write('gradients (numeric):  {}\n'.format(gx_numeric))
            f.write('gradients (backward): {}\n'.format(gx_backward))
            f.write('\n')
            f.write('x: numeric gradient, y: backward gradient')
            f.write(str(e))
>           raise AssertionError(f.getvalue())
E           chainer.testing.function_link.FunctionTestError: backward is not implemented correctly
E           
E           (caused by)
E           AssertionError: check_backward failed (eps=0.001 atol=0.003 rtol=0.01)
E           inputs[0]:
E           array([[0.83642578],
E                  [0.9321289 ],
E                  [-0.55322266]], shape=(3, 1), dtype=float16, device='native:0')
E           inputs[1]:
E           array([[0.65966797, 0.57324219, 0.77246094, 0.97412109],
E                  [-0.21240234, 0.11120605, 0.31933594, 0.86669922],
E                  [-0.034729  , -0.55322266, -0.91357422, 0.11352539]], shape=(3, 4), dtype=float16, device='native:0')
E           grad_outputs[0]:
E           array([[-0.50048828, -0.9868164 , 0.3684082 , 0.29516602],
E                  [0.06579590, 0.7915039 , -0.68554688, -0.921875  ],
E                  [-0.68457031, 0.09558105, 0.02789307, -0.60009766]], shape=(3, 4), dtype=float16, device='native:0')
E           directions[0]:
E           array([[0.43320103],
E                  [-0.14267083],
E                  [0.18341649]], shape=(3, 1), dtype=float64, device='native:0')
E           directions[1]:
E           array([[-0.04748275, -0.54389936, 0.31871587, -0.30096808],
E                  [-0.15560610, 0.13205090, -0.20155451, 0.01276379],
E                  [-0.35331611, 0.20300687, 0.09588751, 0.10208457]], shape=(3, 4), dtype=float64, device='native:0')
E           gradients (numeric):  array(-0.26232969, shape=(), dtype=float64, device='native:0')
E           gradients (backward): array(-0.28092024, shape=(), dtype=float64, device='native:0')
E           
E           x: numeric gradient, y: backward gradient
E           Not equal to tolerance rtol=0.01, atol=0.003
E           
E           (mismatch 100.0%)
E            x: array(-0.26233)
E            y: array(-0.28092)
E           
E           assert_allclose failed: 
E             shape: () ()
E             dtype: float64 float64
E             i: (0,)
E             x[i]: -0.26232968724262784
E             y[i]: -0.28092024120754205
E             relative error[i]: 0.06617733875281573
E             absolute error[i]: 0.018590553964914214
E             relative tolerance * |y[i]|: 0.0028092024120754205
E             absolute tolerance: 0.003
E             total tolerance: 0.00580920241207542
E           x: -0.26232969
E           y: -0.28092024
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:537: FunctionTestError
=============================================== short test summary info ================================================
FAIL tests/chainerx_tests/unit_tests/routines_tests/test_misc.py::test_Maximum_param_174_{in_shapes=((3, 1), (3, 4)), in_dtypes=('float16', 'float16'), input_lhs='random', is_module=False, input_rhs='random', out_dtype='float16'}[native:0]
========================= 1 failed, 88764 passed, 84432 skipped, 41 xfailed in 1327.89 seconds =========================
```

--- a/tests/chainerx_tests/unit_tests/routines_tests/test_misc.py
+++ b/tests/chainerx_tests/unit_tests/routines_tests/test_misc.py
@@ -199,7 +199,13 @@ def test_sign(xp, device, input, dtypes):
 ))
 class TestMaximum(math_utils.BinaryMathTestBase, op_utils.NumpyOpTest):
 
-    dodge_nondifferentiable = True
+    def generate_inputs(self):
+        for _ in range(10):
+            a, b = super().generate_inputs()
+            if ((a.dtype.kind in 'biu' and b.dtype.kind in 'biu') or
+                    (numpy.abs(a - b) > 0.01).all()):
+                return a, b
+        assert False
 
     def func(self, xp, a, b):
         return xp.maximum(a, b)
@@ -303,7 +309,13 @@ class TestMaximumScalar(math_utils.MathScalarTestBase, op_utils.NumpyOpTest):
 ))
 class TestMinimum(math_utils.BinaryMathTestBase, op_utils.NumpyOpTest):
 
-    dodge_nondifferentiable = True
+    def generate_inputs(self):
+        for _ in range(10):
+            a, b = super().generate_inputs()
+            if ((a.dtype.kind in 'biu' and b.dtype.kind in 'biu') or
+                    (numpy.abs(a - b) > 0.01).all()):
+                return a, b
+        assert False
 
     def func(self, xp, a, b):
         return xp.minimum(a, b)

--- a/tests/chainerx_tests/unit_tests/routines_tests/test_misc.py
+++ b/tests/chainerx_tests/unit_tests/routines_tests/test_misc.py
@@ -205,7 +205,7 @@ class TestMaximum(math_utils.BinaryMathTestBase, op_utils.NumpyOpTest):
             if ((a.dtype.kind in 'biu' and b.dtype.kind in 'biu') or
                     (numpy.abs(a - b) > 0.01).all()):
                 return a, b
-        assert False
+        assert False, 'Couldn\'t construct a test case.'
 
     def func(self, xp, a, b):
         return xp.maximum(a, b)
@@ -315,7 +315,7 @@ class TestMinimum(math_utils.BinaryMathTestBase, op_utils.NumpyOpTest):
             if ((a.dtype.kind in 'biu' and b.dtype.kind in 'biu') or
                     (numpy.abs(a - b) > 0.01).all()):
                 return a, b
-        assert False
+        assert False, 'Couldn\'t construct a test case.'
 
     def func(self, xp, a, b):
         return xp.minimum(a, b)

